client/app/scripts/superdesk/editor/editor.js

```diff
@@ -530,6 +543,12 @@ function EditorService(spellcheck, $rootScope, $timeout, $q) {
      * @param {Scope} scope
      */
     this.commitScope = function(scope) {
+        // In case of Kill action, don't undo back to initial value (i.e: original article text) - [ref: SD-3917]
+        // Set initial value to kill template text at the beginning of history.
+        if (scope.$parent.action === 'kill' && scope.history.getIndex() === -1) {
```

Comment:
with this line it looks like we're creating a dependency in editor service for the story. Maybe commitScope function shouldn't be called in the first place.